### PR TITLE
feat(api): add GET data quality stats aggregations endpoint - BED-8611

### PR DESCRIPTION
**Pull request by cami-specter**
## Description

Adds a new `GET /api/v2/data-quality-stats-aggregations` endpoint (BED-8611) that returns  data quality metric counts aggregated per environment kind for an OpenGraph schema  extension, combined across environment instances for each collection run.  
  
The endpoint is behind the `opengraph_extension_management` feature flag.  
  
  **Detailed changes:**
- registers `GET /api/v2/data-quality-stats-aggregations` w/ the FeatureOpenGraphExtensionManagement flag.  
- adds handler supports filtering, sorting, and pagination (GetDataQualityAggregations)
- requires schema_environment_kind_id
- if the schema_extension_id query param is added, it verifies the extension exists and returns 404 if it does not 
- start / end are parsed as plain RFC-3339 timestamps and applied to `created_at`as an inclusive window, like the AZ/AD DQ handlers
- applies hardcoded filtering to match the existing endpoints 
- updates json tags of "schema-"-prefixed fields
- updates sortable columns, and adjusts JSON tags on `DataQualityAggregation`.  
- OpenAPI/Swagger changes  
- adds tests 

## Motivation and Context

Fixes: [BED-8611](https://specterops.atlassian.net/browse/BED-8611)
in support of the Data Quality project

Adds aggregations endpoint for OpenGraph support (we have an existing AD and AZ aggregate endpoint)

## How Has This Been Tested?
- added _Success/_Failure table tests for GetDataQualityAggregations covering basic and edge cases
- ran existing unit tests and integration tests, no regressions
- tested manually via Bruno (see screenshots below), enabling the feature flag locally and by creating mock data for the data_quality_aggregations across multiple environment kinds and extensions.
  
## Screenshots:

    **1. Basic:**
     ?schema_environment_kind_id=182

1_basic

    **2. Pagination:**
     ?schema_environment_kind_id=182&limit=5&skip=0 
     → 5 rows; limit:5, skip:0 are shown, full count.

2_pagination

    **3. Time window:**
     ?schema_environment_kind_id=182&start=2026-07-04T00:00:00Z&end=2026-07-11T00:00:00Z
      → fewer rows.

  
3_time_window

    **4. Ascending sort (non-default):**
     ?schema_environment_kind_id=183&sort_by=-created_at
      → oldest first.

  
4_sorting_ascending(nondefault)

    **5. Extension filter passes (real id):**
     ?schema_environment_kind_id=182&schema_extension_id=3 
     → 200 with rows.

5_extensionfilter1

  **6. Extension filter exists but does not match schema_environment_kind_id:**
     ?schema_environment_kind_id=182&schema_extension_id=4
     → 200; count:0 

5B_mismatch

    **7. Extension does not exist (404):**
     ?schema_environment_kind_id=182&schema_extension_id=999999 
     → 404.

6_extension_id does not exist --  404

    **8. Missing required param (400):**
     ?limit=5 
     → 400 (missing required query parameter: schema_environment_kind_id).

7_missing_required_paramId

  **9. Non-integer required param (400):**
     ?schema_environment_kind_id=abc
     → 400 (query parameter must be an integer: schema_environment_kind_id).

8_noninteger

  **10. neq: rejected for required param (400):**
     ?schema_environment_kind_id=neq:183
     → 400 (query parameter must be an integer: schema_environment_kind_id).

8b_neqnoninteger

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have met the contributing prerequisites
  - Assigned myself to this PR
  - Added the appropriate labels
  - Associated an issue: https://github.com/SpecterOps/BloodHound/issues/672
  - Read the Contributing guide: https://github.com/SpecterOps/BloodHound/wiki/Contributing
- [x] I have ensured that related documentation is up-to-date
  - Open API docs
  - Code comments (GoDocs / JSDocs)
- [x] I have followed proper test practices
  - Added/updated tests to cover my changes
  - All new and existing tests passed

[BED-8611]: https://specterops.atlassian.net/browse/BED-8611?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

* **New Features**
  * Added `GET /api/v2/data-quality-stats-aggregations` to return aggregated data-quality metric counts by environment kind, with optional extension filtering (feature-flagged).

* **Documentation**
  * Updated the OpenAPI spec with the new endpoint, query parameters (`schema_environment_kind_id`, `schema_extension_id`, `start`/`end`, `sort_by`, `skip`/`limit`), and response schema.

* **Bug Fixes**
  * Improved request validation and standardized sorting behavior.
  * Response fields now use `extension_id` and `environment_kind_id`.

* **Tests**
  * Added success and failure tests covering filtering, time windows, pagination, sorting, and error cases.

### CODE REVIEW

File: cmd/api/src/api/v2/dataquality.go [hunk 1]
```diff
@@ -221,6 +223,126 @@ func (s *Resources) GetDataQualityStats(response http.ResponseWriter, request *h
...
+	)
+
+	// omit start and end from parser since they are processed as plain time parsers
+	filterQuery := request.URL.Query()
+	filterQuery.Del("start")
+	filterQuery.Del("end")
+	filterRequest := request.Clone(ctx)
+	filterRequest.URL.RawQuery = filterQuery.Encode()
+
+	queryFilters, err := model.NewQueryParameterFilterParser().ParseQueryParameterFilters(filterRequest)
```
Comment (mvlipka) [inline cmd/api/src/api/v2/dataquality.go]: I don't think we need query param filtering on the endpoint
The existing endpoints don't have it and we don't have an scenario right now where a user would want to query for things such as a `schema_environment_kind_id=nq:<some_id>`

**Review by mvlipka — CHANGES_REQUESTED** [review]

File: cmd/api/src/api/v2/dataquality.go [hunk 2]
```diff
@@ -221,6 +223,112 @@ func (s *Resources) GetDataQualityStats(response http.ResponseWriter, request *h
...
+
+	// schema_environment_kind_id is required
+	kindID := queryParams.Get(schemaEnvironmentKindIDParam)
+	if kindID == "" {
+		api.WriteErrorResponse(ctx, api.BuildErrorResponse(http.StatusBadRequest,
+			fmt.Sprintf(api.FmtErrorResponseDetailsMissingRequiredQueryParameter, schemaEnvironmentKindIDParam), request),
+			response)
+		return
+	}
+	if _, err := strconv.ParseInt(kindID, 10, 32); err != nil {
```
[the remark this answers is not in the document]
Comment (cami-specter) [reply cmd/api/src/api/v2/dataquality.go:252]: This is intentional, just made this change. Which public contract? The AD/AZ endpoints all use plain integers ans well. I updated the OpenAPI yaml (data-quality.data-quality-stats-aggregations.yaml) and regenerated the openapi.json to reflect that as well.

File: packages/go/openapi/src/paths/data-quality.ad-domains.id.data-quality-stats.yaml [hunk 3]
```diff
@@ -44,7 +44,7 @@ get:
         type: string
         format: date-time
     - name: end
-      description: Ending datetime of range (exclusive) in RFC-3339 format; Defaults
+      description: Ending datetime of range (inclusive) in RFC-3339 format; Defaults
```
Comment (cami-specter) [inline packages/go/openapi/src/paths/data-quality.ad-domains.id.data-quality-stats.yaml:47]: Fixing the OpenAPI docs here and for the rest of the older AD/AZ endpoints because they keep both start and end inclusive (created_at is between start and end: for example [here](https://github.com/SpecterOps/BloodHound/blob/main/cmd/api/src/database/dataquality.go#L58-L76) and  [here](https://github.com/SpecterOps/BloodHound/blob/main/cmd/api/src/database/dataquality.go#L256-L265))

eg
```
const (
		defaultWhere = "created_at between ? and ?"
	)
```

List of all time windows:
-  [GetADDataQualityStats](https://github.com/SpecterOps/BloodHound/blob/main/cmd/api/src/database/dataquality.go#L58)
- [GetAzureDataQualityStats](https://github.com/SpecterOps/BloodHound/blob/main/cmd/api/src/database/dataquality.go#L159)
- [GetADDataQualityAggregations](https://github.com/SpecterOps/BloodHound/blob/main/cmd/api/src/database/dataquality.go#L256)
- [GetAzureDataQualityAggregations](https://github.com/SpecterOps/BloodHound/blob/main/cmd/api/src/database/dataquality.go#L289)

File: cmd/api/src/api/v2/dataquality.go [hunk 4]
```diff
@@ -37,12 +38,13 @@ import (
...
-	ErrNoEnvironmentId   string = "environment_id is required"
-	ErrUnknownUser       string = "unknown user"
-	ErrNoAccess          string = "user does not have permission to access this environment"
+	ErrNoAccess                        string = "user does not have permission to access this environment"
+	ErrNoEnvironmentId                 string = "environment_id is required"
+	ErrNoTenantId                      string = "no tenant id specified in url"
+	ErrNoPlatformId                    string = "no platform id specified in url"
+	ErrUnknownUser                     string = "unknown user"
+	FmtErrInvalidIntegerQueryParameter string = "query parameter must be an integer: %s"
+	FmtErrInvalidPlatformId            string = "invalid platform id specified in url: %v"
```
Comment (mvlipka) [inline cmd/api/src/api/v2/dataquality.go:47]: We should keep these with the standard `Error` prefix

**Review by mvlipka — CHANGES_REQUESTED** [review]
Looks good! Tested and verified the results through Bruno
Just gotta fix that nit to conform to our standard

File: cmd/api/src/api/v2/dataquality.go (hunk 4 above)
Comment (mvlipka) [reply cmd/api/src/api/v2/dataquality.go:47]: My mistake, I missed that this was being used in a fmt.Sprintf
Good to go!

**Review by mvlipka — APPROVED** [review]

File: cmd/api/src/api/v2/dataquality.go (hunk 4 above)
Comment (cami-specter) [reply cmd/api/src/api/v2/dataquality.go:47]: Thank you!
